29th_first_semina/app/src/main/java/com/example/a29th_first_semina/SignInActivity.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.a29th_first_semina
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import com.example.a29th_first_semina.databinding.ActivitySigninBinding
+
+class SignInActivity : AppCompatActivity() {
+    private lateinit var binding: ActivitySigninBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySigninBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        initButtonClickEvent()
+
+    }
+
+    private fun initButtonClickEvent() {
+
+        clickLoginEvent()
+        clickSignUpEvent()
+    }
+
+    private fun moveHomeActivity() {
+        val intent = Intent(this, HomeActivity::class.java)
+        startActivity(intent)
+    }
+
+    private fun moveSignUpActivity() {
+        val signupintent = Intent(this, SignUpActivity::class.java)
+        startActivity(signupintent)
+
+    }
+
+    private fun clickLoginEvent() {
+
+        binding.apply {
+            btnLogin.setOnClickListener {
+                val userId: String = etId.text.toString()
+                val userPwd: String = etPwd.text.toString()
```

Comment:
코틀린에서는 타입추론 기능을 지원하기 때문에
```suggestion
                val userId = etId.text.toString()
                val userPwd = etPwd.text.toString()
```
이런식으로 작성하셔도 괜찮습니다~!!

Reply:
그리고 아래 코드를 봤는데 toString()의 경우 빼주셔도 괜찮을 것 같아요!!